entity-framework/efcore-and-ef6/porting/port-behavior.md

```diff
@@ -17,13 +17,15 @@ In EF6, calling `DbSet.Add()` on an entity results in a recursive search for all
 
```

Comment:
Couple of comments:
- This change of behavior only applies to the `Attach` and `Update` method groups. `Add` always puts entities in the `Added` state, even if the key is set.
- `Attach` methods put entities with keys set into the `Unchanged` state. This facilitates "insert it if new, otherwise leave it alone." `Update` methods put entities with keys set into the `Modified` state. This facilitates "insert it if new, otherwise update it."

The general philosophy here is that `Update` is a very simple way to handle inserts and updates of disconnected entities. It ensures any new entities are inserted, and any existing entities are updated.

At the same time, `Add` still provides an easy way force entities to be inserted. `Add` is mostly useful only when not using store-generated keys, such that EF doesn't know if the entity is new or not.